s3/src/main/java/io/ddf/s3/S3DDFManager.java

```diff
@@ -0,0 +1,235 @@
+package io.ddf.s3;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.*;
+
+import io.ddf.DDF;
+import io.ddf.DDFManager;
+
+import io.ddf.datasource.DataFormat;
+import io.ddf.datasource.DataSourceDescriptor;
+import io.ddf.datasource.S3DataSourceCredentials;
+import io.ddf.datasource.S3DataSourceDescriptor;
+import io.ddf.exception.DDFException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Created by jing on 12/2/15.
+ */
+
+public class S3DDFManager extends DDFManager {
+    // Descriptor.
+    private S3DataSourceCredentials mCredential;
+    // Amazon client connection.
+    private AmazonS3 mConn;
+    // Upper limit for content preview.
+    final int K_LIMIT = 1000;
+
+
+    // TODO: Remove Engine Type
+    public S3DDFManager(S3DataSourceDescriptor s3dsd, EngineType engineType) throws DDFException {
+
+        mCredential = (S3DataSourceCredentials)s3dsd.getDataSourceCredentials();
```

Comment:
The `S3DataSourceDescriptor` should use interface specialization to return `S3DataSourceCredentials`.